core/logger/zap.go

```diff
@@ -98,6 +189,10 @@ func (l *zapLogger) ErrorIfClosing(c io.Closer, name string) {
 }
 
 func (l *zapLogger) Sync() error {
+	if l.config.local.ToDisk {
+		l.closeDiskPollChan <- struct{}{}
+	}
```

Comment:
`Sync()` is normally allowed to be called multiple times. If we are only using it to clean up though, then maybe rename it to `Close()`?  Otherwise, I think this part needs to be moved to a new `Close()` method separately.

Reply:
Actually, having Close() on the interface seems problematic. I think we'll have to do something like return a `func close()` from the constructor that spawns the goroutine instead.